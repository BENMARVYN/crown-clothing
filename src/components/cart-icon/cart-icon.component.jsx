import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect'

import { toggleCartHidden } from '../../redux/cart/cart.action';
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';

import { ReactComponent as ShoppingIcon } from '../../assets/shopping-bag.svg';

import { CartIconContainer, ShoppingIconContainer, ItemCount } from './cart-icon.styles';

const CartIcon = ({ toggleCartHidden, itemCount }) => (
  <CartIconContainer onClick={toggleCartHidden}>
    <ShoppingIconContainer>
      <ShoppingIcon />
    </ShoppingIconContainer>
    <ItemCount>{ itemCount }</ItemCount>
  </CartIconContainer>
);

const matchStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount
});

const mapDispatchToProps = dispatch => ({
  toggleCartHidden: () => dispatch(toggleCartHidden())
});

export default connect(matchStateToProps, mapDispatchToProps)(CartIcon);