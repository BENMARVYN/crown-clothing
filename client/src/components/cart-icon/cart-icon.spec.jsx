import React from "react";
import { shallow } from "enzyme";
import { CartIcon } from "./cart-icon.component";

let mockToggleCartHidden;

beforeEach(() => {
  mockToggleCartHidden = jest.fn();
});

describe("Testing CartIcon component", () => {
  it("Should match snapshot", () => {
    expect(
      shallow(
        <CartIcon toggleCartHidden={mockToggleCartHidden} itemCount={0} />
      )
    ).toMatchSnapshot();
  });
});
