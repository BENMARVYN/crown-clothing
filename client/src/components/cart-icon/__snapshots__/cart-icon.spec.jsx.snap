// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CartIcon component Should match snapshot 1`] = `ShallowWrapper {}`;
